### PR TITLE
feat(stdlib): Add ASCII utilities to String, Char

### PR DESCRIPTION
**Pull request by alex-snezhko**
Add ASCII utilities to `String` and `Char` modules

### CODE REVIEW

**Review by spotandjake — COMMENTED** [review]
This looks good to me. Just needs docs generated.

**Comment by alex-snezhko** [comment]
Generated

**Review by spotandjake — APPROVED** [review]

File: stdlib/char.gr [hunk 1]
```diff
@@ -259,3 +259,57 @@ provide let (>=) = (x: Char, y: Char) => {
   let y = WasmI32.fromGrain(y)
   x >= y
 }
+
+/**
+ * Checks if the character is a digit.
```
Comment (ospencer) [inline stdlib/char.gr:264]:
```suggestion
 * Checks if the character is an ASCII digit.
```

File: stdlib/char.gr [hunk 2]
```diff
@@ -259,3 +259,57 @@ provide let (>=) = (x: Char, y: Char) => {
...
+ *
+ * @param char: The character
+ * @returns `true` if the character is a digit or `false` otherwise
```
Comment (ospencer) [inline stdlib/char.gr:267]:
```suggestion
 * @returns `true` if the character is an ASCII digit or `false` otherwise
```

**Review by ospencer — APPROVED** [review]

File: compiler/test/stdlib/char.test.gr [hunk 3]
```diff
@@ -86,3 +86,26 @@ assert 'Z' >= 'A'
 assert !('a' >= 'b')
 assert 'a' >= 'a'
 assert 'B' >= 'B'
+
+// isAsciiDigit
+assert Char.isAsciiDigit('1')
+assert !Char.isAsciiDigit('a')
+assert !Char.isAsciiAlpha('🌾')
```
Comment (phated) [inline compiler/test/stdlib/char.test.gr:93]: These are the ascii digit tests according to the comment

File: stdlib/char.gr [hunk 4]
```diff
@@ -314,3 +314,57 @@ provide let (>=) = (x: Char, y: Char) => {
   let y = WasmI32.fromGrain(y)
   x >= y
 }
+
+/**
+ * Checks if the character is an ASCII digit.
+ *
+ * @param char: The character
```
Comment (phated) [inline stdlib/char.gr:321]: We generally write these like: "The character to check"

File: stdlib/char.gr [hunk 5]
```diff
@@ -314,3 +314,57 @@ provide let (>=) = (x: Char, y: Char) => {
...
+ * @returns `true` if the character is an ASCII digit or `false` otherwise
+ *
+ * @since v0.6.0
+ */
+provide let isAsciiDigit = char => char >= '0' && char <= '9'
+
+/**
+ * Checks if the character is an ASCII alphabetical character.
+ *
+ * @param char: The character
```
Comment (phated) [inline stdlib/char.gr:331]: See above

File: stdlib/char.gr [hunk 6]
```diff
@@ -314,3 +314,57 @@ provide let (>=) = (x: Char, y: Char) => {
...
+ * @since v0.6.0
+ */
+provide let isAsciiAlpha = char =>
+  char >= 'a' && char <= 'z' ||
+    char >= 'A' && char <= 'Z'
+
+/**
+ * Converts the character to ASCII lowercase if it is an ASCII uppercase character.
+ * 
+ * @param char: The character
```
Comment (phated) [inline stdlib/char.gr:343]: "The character to convert"?

"The character to transform"?

File: stdlib/char.gr [hunk 7]
```diff
@@ -314,3 +314,57 @@ provide let (>=) = (x: Char, y: Char) => {
...
+    fromCode(code(char) + 0x20)
+  } else {
+    char
+  }
+}
+
+/**
+ * Converts the character to ASCII uppercase if it is an ASCII lowercase character.
+ * 
+ * @param char: The character
```
Comment (phated) [inline stdlib/char.gr:359]: See above

File: stdlib/string.gr [hunk 8]
```diff
@@ -2150,3 +2151,37 @@ provide let trim = (string: String) => {
   )
   return WasmI32.toGrain(str): String
 }
+
+/**
+ * Converts all ASCII uppercase characters in the string to lowercase.
+ * 
+ * @param string: The string
```
Comment (phated) [inline stdlib/string.gr:2158]: See previous comments

File: stdlib/string.gr [hunk 9]
```diff
@@ -2150,3 +2151,37 @@ provide let trim = (string: String) => {
...
+  for (let mut i = 0; i < len; i += 1) {
+    chars[i] = Char.toAsciiLowercase(chars[i])
+  }
+  implode(chars)
+}
+
+/**
+ * Converts all ASCII lowercase characters in the string to uppercase.
+ * 
+ * @param string: The string
```
Comment (phated) [inline stdlib/string.gr:2175]: See previous comments

**Review by phated — CHANGES_REQUESTED** [review]
Doc comments and one test, but otherwise looks good

**Comment by phated** [comment]
Needs regen too, I think

**Review by phated — APPROVED** [review]

**Pull request merged**